### PR TITLE
Apply BestieTemplate v0.12.0

### LINKED ISSUE #676: Apply BestieTemplate.jl
### What and Why

The [BestieTemplate.jl](https://github.com/abelsiqueira/BestieTemplate.jl) was created using the best practices that we introduced in this package, so for many parts, this package is up-to-date in those practices.
However, since it was created, there were improvements in the BestieTemplate that we should apply here and the easier way to do it is by adopting the template.

The template was applied to [TulipaIO](https://github.com/TulipaEnergy/TulipaIO.jl) recently (in two parts, check the merged PRs). It was named COPIERTemplate.jl at the moment.

Some of the things:

- Link checker was removed from pre-commit and moved to a separate Linting with a different faster package (lychee)
- Dependabot was added (#444)
- An action checking for updates of the pre-commit versions was added
- It helps with this checklist (#630)
- Better solution for testing on the PR 
- Maybe [All Contributors](https://allcontributors.org)

If we do decide to apply it, we should wait until at least Monday, when the Registration of the rename will be done.

### Possible Drawbacks

There are different choices, such as the issue and PR templates are different, the formatting is slightly different, and the README.dev and CONTRIBUTING files moved to the docs.

It takes some time to apply it the first time, since you have to fix the conflicts (though I can help with that).

I am biased, since I'm lead on the project.

### Related Issues

_No response_

### PR DESCRIPTION
**Pull request by abelsiqueira**
# Pull request details

## Describe the changes made in this pull request

Explicitly make Julia 1.6 the minimum supported version
Apply BestieTemplate v0.12.0

## List of related issues or pull requests

Closes #676

## Collaboration confirmation

As a contributor I confirm

-   [x] I read and followed the instructions in README.dev.md
-   [x] The documentation is up to date with the changes introduced in this Pull Request (or NA)
-   [x] Tests are passing
-   [x] Lint is passing

### CODE REVIEW

File: .github/workflows/TestOnPRs.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  # Cancel intermediate builds: only if it is a pull request build.
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
+
+jobs:
+  test:
+    uses: ./.github/workflows/ReusableTest.yml
+    with:
+      os: ubuntu-latest
+      version: "1"
```
Comment (datejada) [inline .github/workflows/TestOnPRs.yml:24]:
```suggestion
      version: "lts"
```

File: .github/workflows/TestOnPR.yml [hunk 2]
```diff
@@ -2,45 +2,45 @@
...
-                  files: lcov.info
-                  token: ${{ secrets.CODECOV_TOKEN }} # Defined in the organization secrets
+  test:
+    name: Julia ${{ matrix.version }} - ${{ matrix.os }} - ${{ matrix.arch }} - ${{ github.event_name }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        version:
+          - "1.6"
```
Comment (datejada) [inline .github/workflows/TestOnPR.yml:25]:
```suggestion
          - "lts"
```

File: .github/workflows/TestOnPR.yml
Comment (datejada) [inline .github/workflows/TestOnPR.yml:1]: Do we need this file? I think Bestie uses now TestOnPRs.yml instead (just checking)

**Review by datejada — COMMENTED** [review]
Hi @abelsiqueira some minor comments so far. Thanks!

File: .github/workflows/TestOnPR.yml
Comment (abelsiqueira) [reply .github/workflows/TestOnPR.yml:1]: We must have forgotten to delete

File: .github/workflows/TestOnPRs.yml (hunk 1 above)
Comment (abelsiqueira) [reply .github/workflows/TestOnPRs.yml:24]: This we keep as "1", to test only on the latest release (1.11)

Comment (datejada) [reply .github/workflows/TestOnPRs.yml:24]: So, should we test in both? "1" and "lts" since this is how we are doing the test on PRs currently

**Review by datejada — APPROVED** [review]
LGTM

**Pull request merged**